### PR TITLE
Fix extra submission form in delayed feedback

### LINKED ISSUE #935: The response dialog of a submit exercise shows another submission form with delayed feedback
When the ``reveal-submission-feedback`` is used and a student submits a programming exercise, A+ shows another submission form in the response dialog.

```
.. submit:: class_y_programming A600
  :config: exercises/programming/exam-y/config.yaml
  :radar_tokenizer: python
  :title: Extending the program code
  :submissions: 5
  :reveal-submission-feedback: deadline +75min
```

delayed-feedback-bug

I think this is more than a cosmetic bug, because the student can resubmit their file using the *submission form inside the response dialog*. Thus the student can submit the same file twice by mistake.

I encountered this when testing A+ locally (containers *apluslms/run-aplus-front:1.11* and *apluslms/run-mooc-grader:1.11*).

### PR DESCRIPTION
**Pull request by ihalaij1**
# Description

**What?**

Fix extra submission form appearing in delayed feedback with file submit exercises.
This PR also changes ``submission.html``to use tabs instead of spaces for indentation, as is agreed with the A+ style guide.

**Why?**

This issue was reported by a teacher.
The extra form along with its file input and submit-button was confusing to students and could lead to wasted submissions.

**How?**

File submit exercises no longer show the exercise instructions along with the file input field and submit-button in the delayed feedback before the feedback is given.

Fixes #935

# Testing

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [x] Manual testing.

Tested that delayed feedback no longer has the confusing layout in the submission modal or plain submission page.

**Did you test the changes in**

- [x] Chrome
- [ ] Firefox
- [ ] This pull request cannot be tested in the browser.

# Translation

- [ ] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [x] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
- [ ] Did you use Python type hinting in all functions that you added or edited? ([type hints](https://docs.python.org/3/library/typing.html) for function parameters and return values)

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

### CODE REVIEW

**Review by Mankro — CHANGES_REQUESTED** [review]
Great and simple approach! I was thinking of changes so that we could avoid a database query for checking the existence of submitted files and also to support delayed feedback in questionnaires that contain both multiple-choice questions and file upload fields. What do you think of the ideas below? Is it going to work?

During testing, I realized that the code execution paths are different for opening the feedback modal from the 'My submissions' dropdown and for displaying the feedback modal right after submitting a file upload assignment (questionnaires do not show any modal right after submitting). These need to be tested carefully.

In the future, when you fix the indentation in old code files, it would be best to make separate commits for fixing the indentation and making functional changes in the code. It would be easier to read the git diff since the indentation changes add a lot of noise there.

**Comment by ihalaij1** [comment]
I made some changes based on your suggestions. Tell me what you think about this approach.
The exercise descriptions are now visible in the delayed feedback, with file input fields disabled and submit-button removed.

File: exercise/protocol/exercise_page.py [hunk 1]
```diff
@@ -81,17 +82,21 @@ def _find_exercise_element(self, soup: BeautifulSoup) -> Optional[Tag]:
...
         exercise element.
         """
+        force_disable_submit = False
         # Find all form elements on the exercise page and fill in the values
         field_elements = form_element.find_all(['input', 'select', 'textarea'])
         for field_element in field_elements:
             field_element = cast(Tag, field_element)
             field_name = cast(str, field_element.get('name'))
             if field_name not in field_values:
+                if field_element.name == 'input' and field_element.get('type') == 'file':
```
[the remark this answers is not in the document]
Comment (ihalaij1) [reply exercise/protocol/exercise_page.py]: I made changes to the logic.

The code now checks if the feedback of the exercise is revealed by checking the active ``RevealRule``.
With SubmissionDrafts the new variable ``feedback_revealed`` defaults to ``True``, and the submit-button and file input fields are therefore not disabled (even though no actual exercise feedback is actually revealed, of course).

**Review by Mankro — CHANGES_REQUESTED** [review]
Excellent! I tested this too and it works in almost all cases. I also thought we can optimize `feedback_revealed` in `Submission.load()` because the value already exists in the view classes and they could just pass the value as a parameter to `Submission.load()`. I implemented this change myself.

The case that does not work:
* questionnaire that contains file upload fields and `data-aplus-quiz` attribute in the exercise html `<div>` element
* feedback is currently hidden due to delayed feedback (`exercise.submission_feedback_reveal_rule`)
* the submit button disappears after submitting and the student can not submit again
* -> Let's ignore this now since it does not affect any current courses!

**Review by Mankro — APPROVED** [review]

File: exercise/submission_models.py [hunk 2]
```diff
@@ -445,6 +445,7 @@ def load(self, request: HttpRequest, allow_submit: bool = True) -> ExercisePage:
         The `allow_submit` argument determines if the submit button will be
         shown on the page.
         """
+        from .reveal_states import ExerciseRevealState
```
[the remark this answers is not in the document]
Comment (ihalaij1) [reply exercise/submission_models.py]: That was due to a circular import.

**Pull request merged**